exec/internal/compile/scanner.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2019 The go-interpreter Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package compile
+
+import (
+	ops "github.com/go-interpreter/wagon/wasm/operators"
+)
+
+type scanner struct {
+	supportedOpcodes map[byte]bool
```

Comment:
depending on the expected size of this map, we might want to consider replacing it with `map[byte]struct{}` (that is slightly leaner in terms of memory usage.)

Reply:
ok.